src/components/PeoplePicker/PeoplePicker.tsx

```diff
@@ -0,0 +1,238 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import { IPeoplePickerProps } from './PeoplePicker.Props';
+import { autobind } from '../../utilities/autobind';
+import './PeoplePicker.scss';
+import { Principal } from './Principal';
+import { Label } from './../../../src/components/Label/Label';
+import { IPrincipal } from './Principal.Props';
+import { Spinner } from '../Spinner/Spinner';
+import { SpinnerType } from '../Spinner/Spinner.Props';
+import { Tooltip } from '../Tooltip/Tooltip';
+import { DirectionalHint } from '../../utilities/DirectionalHint';
+import { Icon } from '../Icon/Icon';
+
+export interface IPeoplePickerState {
+    isFocused?: boolean;
+    selectedPrincipalList?: IPrincipal[];
+    suggestionsVisible?: boolean;
+    value?: any;
+}
+
+export class PeoplePicker extends React.PureComponent<IPeoplePickerProps, IPeoplePickerState> {
+    private _field;
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isFocused: false,
+            selectedPrincipalList: null,
+            suggestionsVisible: false,
+            value: props.value || props.defaultValue || ''
+        };
+    }
+
+    @autobind
+    private _onInputChange(event: React.ChangeEvent<any>): void {
+        const element: HTMLTextAreaElement = event.target as HTMLTextAreaElement;
+        const value: any = element.value;
+
+        this.setState({
+            value: value
+        });
+
+        if (value.length >= 3) {
+            this.setState({ suggestionsVisible: true });
+            this.props.onSearch(value);
+        } else {
+            this.setState({ suggestionsVisible: false });
+        }
+    }
+
+    private _renderSuggestions(): JSX.Element {
+        return (
+            <div className="people-picker-suggestions">
+                {this.props.loadingSuggestionList && <Spinner type={SpinnerType.small} />}
+                {!this.props.loadingSuggestionList && this.props.suggestionList.map((principal, index) => (
+                    <Principal
+                        principal={principal}
+                        isSelected={false}
+                        onSelect={this._onSuggestionClick}
+                    />
+                ))}
+            </div>
+        );
+    }
+
+    @autobind
+    private _onSuggestionClick(principal: IPrincipal) {
+        this.setState({ suggestionsVisible: false, value: '' });
+
+        if (this.state.selectedPrincipalList !== null) {
+            this.setState({ selectedPrincipalList: [...this.state.selectedPrincipalList, principal] });
+        } else {
+            this.setState({ selectedPrincipalList: [principal] });
+        }
+    }
+
+    @autobind _ref(value: HTMLInputElement) {
```

Comment:
Can you format this so the ref function is in new line? also declare it private _ref.

Reply:
yes :) I did it